### PR TITLE
fs_heap: added support for initialization from a specified section

### PR DESCRIPTION
**Pull request by crafcat7**
## Summary
**Why is this change necessary?**
This PR is to enhance the use of fs_heap.

**Changes**
1. Based on the changes in https://github.com/apache/nuttx/pull/13150, I added a new way to initialize fs_heap by allocating memory through the specified section. This method requires modifying the ld file to allocate a space for fs to use
2. Replace all kmm_ application codes in nuttx/fs with fs_heap. By default, it is equivalent to kmm_
3. Move accept4 from net to fs because it uses sockfd_allocate, which will cause its priv to be released in fs.

## Impact
**Is a new feature added?:** NO
**Impact on build:** NO
**Impact on hardware:** NO, this change does not specifically target any particular hardware architectures or boards.
**Impact on documentation:** NO,This patch does not introduce any new features
**Impact on compatibility:** This implementation aims to be backward compatible. No existing functionality is expected to be broken.

## Testing
Build Host(s): Linux x86
Target(s): sim/nsh
Based on the relevant fs case is passed (Because fs_heap_ is equivalent to kmm_ by default, there is no impact on any existing configurations.)

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: fs/mmap/fs_rammap.c [hunk 1]
```diff
@@ -263,7 +263,8 @@ int rammap(FAR struct file *filep, FAR struct mm_map_entry_s *entry,
 
   /* Allocate a region of memory of the specified size */
 
-  rdbuffer = type == MAP_KERNEL ? fs_heap_malloc(length) : kumm_malloc(length);
+  rdbuffer = type == MAP_KERNEL ? fs_heap_malloc(length)
+                                : kumm_malloc(length);
```
Comment (anchao) [inline fs/mmap/fs_rammap.c:267]: Squash the last change

File: fs/fat/fs_fat32.c [hunk 2]
```diff
@@ -469,7 +469,7 @@ static int fat_close(FAR struct file *filep)
 
   /* Then free the file structure itself. */
 
-  kmm_free(ff);
+  fs_heap_free(ff);
```
Comment (anchao) [inline fs/fat/fs_fat32.c:472]: `fs/fat/fs_fat32.c:507:  buf = kmm_malloc(fs->fs_hwsectorsize);`

File: fs/procfs/fs_procfspressure.c [hunk 3]
```diff
@@ -121,7 +123,7 @@ static int pressure_open(FAR struct file *filep, FAR const char *relpath,
       return -ENOENT;
     }
 
-  priv = kmm_zalloc(sizeof(struct pressure_file_s));
+  priv = fs_heap_zalloc(sizeof(struct pressure_file_s));
```
Comment (anchao) [inline fs/procfs/fs_procfspressure.c:126]: image

File: fs/mmap/fs_rammap.c (hunk 1 above)
Comment (crafcat7) [reply fs/mmap/fs_rammap.c:267]: Done

File: fs/procfs/fs_procfspressure.c (hunk 3 above)
Comment (crafcat7) [reply fs/procfs/fs_procfspressure.c:126]: Done

File: fs/fat/fs_fat32.c (hunk 2 above)
Comment (crafcat7) [reply fs/fat/fs_fat32.c:472]: Done

**Review by anchao — APPROVED** [review]

**Pull request merged**